{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Variant Calling\n",
    "\n",
    "Now that we have the aligned sequences to the reference, we can look for difference between the aligned sequences and the reference. This process is also known as variant calling.\n",
    "\n",
    "Let's review the files that we have."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 482M\n",
      "-rw-r--r-- 1 root root  15K Jul 24 09:02 '01 - Preparations for Finding a Disease Mutation.ipynb'\n",
      "-rw-r--r-- 1 root root  17K Jul 24 11:07 '02 - Aligning the FASTQ File.ipynb'\n",
      "-rw-r--r-- 1 root root  60K Jul 24 08:55 '03 - Variant Calling.ipynb'\n",
      "-rw-r--r-- 1 root root  11K Jul 24 08:52 '04 - Annotation of Variants.ipynb'\n",
      "-rw-r--r-- 1 root root 177M Jul 24 08:53  chr5.fa\n",
      "-rw-r--r-- 1 root root  588 Jul 24 08:59  chr5.fa.amb\n",
      "-rw-r--r-- 1 root root   44 Jul 24 08:59  chr5.fa.ann\n",
      "-rw-r--r-- 1 root root 174M Jul 24 08:59  chr5.fa.bwt\n",
      "-rw-r--r-- 1 root root   23 Jul 24 08:54  chr5.fa.fai\n",
      "-rw-r--r-- 1 root root  44M Jul 24 08:59  chr5.fa.pac\n",
      "-rw-r--r-- 1 root root  87M Jul 24 09:00  chr5.fa.sa\n",
      "-rw-r--r-- 1 root root 820K Jul 24 08:52  input.fq\n",
      "-rw-r--r-- 1 root root 225K Jul 24 08:54  input_fastqc.html\n",
      "-rw-r--r-- 1 root root 235K Jul 24 08:54  input_fastqc.zip\n",
      "-rw-r--r-- 1 root root    0 Jul 24 08:54  mapped.bam\n",
      "-rw-r--r-- 1 root root 963K Jul 24 11:06  mapped.sam\n"
     ]
    }
   ],
   "source": [
    "ls -lh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Preparing for variant calling\n",
    "\n",
    "In order to perform variant calling, we need to preprocess the SAM file as well as the reference chromosome 5 fasta file. The following steps will be performed :\n",
    "\n",
    "* Converting SAM to BAM format (this is the binary compressed version)\n",
    "* Sorting the BAM file\n",
    "* Indexing the BAM file\n",
    "* Indexing the reference fasta file\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Processing the SAM file generated by BWA\n",
    "\n",
    "We will process the SAM file using samtools, which can be accessed after loading the module."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Usage: samtools view [options] <in.bam>|<in.sam>|<in.cram> [region ...]\n",
      "\n",
      "Options:\n",
      "  -b       output BAM\n",
      "  -C       output CRAM (requires -T)\n",
      "  -1       use fast BAM compression (implies -b)\n",
      "  -u       uncompressed BAM output (implies -b)\n",
      "  -h       include header in SAM output\n",
      "  -H       print SAM header only (no alignments)\n",
      "  -c       print only the count of matching records\n",
      "  -o FILE  output file name [stdout]\n",
      "  -U FILE  output reads not selected by filters to FILE [null]\n",
      "  -t FILE  FILE listing reference names and lengths (see long help) [null]\n",
      "  -L FILE  only include reads overlapping this BED FILE [null]\n",
      "  -r STR   only include reads in read group STR [null]\n",
      "  -R FILE  only include reads with read group listed in FILE [null]\n",
      "  -q INT   only include reads with mapping quality >= INT [0]\n",
      "  -l STR   only include reads in library STR [null]\n",
      "  -m INT   only include reads with number of CIGAR operations consuming\n",
      "           query sequence >= INT [0]\n",
      "  -f INT   only include reads with all  of the FLAGs in INT present [0]\n",
      "  -F INT   only include reads with none of the FLAGS in INT present [0]\n",
      "  -G INT   only EXCLUDE reads with all  of the FLAGs in INT present [0]\n",
      "  -s FLOAT subsample reads (given INT.FRAC option value, 0.FRAC is the\n",
      "           fraction of templates/read pairs to keep; INT part sets seed)\n",
      "  -M       use the multi-region iterator (increases the speed, removes\n",
      "           duplicates and outputs the reads as they are ordered in the file)\n",
      "  -x STR   read tag to strip (repeatable) [null]\n",
      "  -B       collapse the backward CIGAR operation\n",
      "  -?       print long help, including note about region specification\n",
      "  -S       ignored (input format is auto-detected)\n",
      "      --input-fmt-option OPT[=VAL]\n",
      "               Specify a single input file format option in the form\n",
      "               of OPTION or OPTION=VALUE\n",
      "  -O, --output-fmt FORMAT[,OPT[=VAL]]...\n",
      "               Specify output format (SAM, BAM, CRAM)\n",
      "      --output-fmt-option OPT[=VAL]\n",
      "               Specify a single output file format option in the form\n",
      "               of OPTION or OPTION=VALUE\n",
      "  -T, --reference FILE\n",
      "               Reference sequence FASTA FILE [null]\n",
      "  -@, --threads INT\n",
      "               Number of additional threads to use [0]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "samtools view"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will begin by converting the SAM file to the compressed BAM format. To do this, we will need to refer to the reference fasta file and the SAM file. The output will be redirected to a new BAM file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [],
   "source": [
    "samtools view -bT chr5.fa mapped.sam > mapped.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 483M\n",
      "-rw-r--r-- 1 root root  15K Jul 24 09:02 '01 - Preparations for Finding a Disease Mutation.ipynb'\n",
      "-rw-r--r-- 1 root root  17K Jul 24 11:07 '02 - Aligning the FASTQ File.ipynb'\n",
      "-rw-r--r-- 1 root root  61K Jul 24 11:38 '03 - Variant Calling.ipynb'\n",
      "-rw-r--r-- 1 root root  11K Jul 24 08:52 '04 - Annotation of Variants.ipynb'\n",
      "-rw-r--r-- 1 root root 177M Jul 24 08:53  chr5.fa\n",
      "-rw-r--r-- 1 root root  588 Jul 24 08:59  chr5.fa.amb\n",
      "-rw-r--r-- 1 root root   44 Jul 24 08:59  chr5.fa.ann\n",
      "-rw-r--r-- 1 root root 174M Jul 24 08:59  chr5.fa.bwt\n",
      "-rw-r--r-- 1 root root   23 Jul 24 08:54  chr5.fa.fai\n",
      "-rw-r--r-- 1 root root  44M Jul 24 08:59  chr5.fa.pac\n",
      "-rw-r--r-- 1 root root  87M Jul 24 09:00  chr5.fa.sa\n",
      "-rw-r--r-- 1 root root 820K Jul 24 08:52  input.fq\n",
      "-rw-r--r-- 1 root root 225K Jul 24 08:54  input_fastqc.html\n",
      "-rw-r--r-- 1 root root 235K Jul 24 08:54  input_fastqc.zip\n",
      "-rw-r--r-- 1 root root 212K Jul 24 11:38  mapped.bam\n",
      "-rw-r--r-- 1 root root 963K Jul 24 11:06  mapped.sam\n"
     ]
    }
   ],
   "source": [
    "ls -lh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice that the BAM files is much smaller than the SAM file\n",
    "\n",
    "Next we will mark any duplicates in alignments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "finding positions of the duplicate reads in the file...\n",
      "  sorted 0 end pairs\n",
      "     and 3712 single ends (among them 0 unmatched pairs)\n",
      "  collecting indices of duplicate reads...   done in 3 ms\n",
      "  found 1028 duplicates\n",
      "collected list of positions in 0 min 0 sec\n",
      "marking duplicates...\n",
      "total time elapsed: 0 min 0 sec\n"
     ]
    }
   ],
   "source": [
    "sambamba markdup mapped.bam mapped.dedup.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next, we will sort and index the BAM file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [],
   "source": [
    "samtools sort -o mapped.dedup.sort.bam mapped.dedup.bam \n",
    "samtools index mapped.dedup.sort.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 483M\n",
      "-rw-r--r-- 1 root root  15K Jul 24 09:02 '01 - Preparations for Finding a Disease Mutation.ipynb'\n",
      "-rw-r--r-- 1 root root  17K Jul 24 11:07 '02 - Aligning the FASTQ File.ipynb'\n",
      "-rw-r--r-- 1 root root  62K Jul 24 11:42 '03 - Variant Calling.ipynb'\n",
      "-rw-r--r-- 1 root root  11K Jul 24 08:52 '04 - Annotation of Variants.ipynb'\n",
      "-rw-r--r-- 1 root root 177M Jul 24 08:53  chr5.fa\n",
      "-rw-r--r-- 1 root root  588 Jul 24 08:59  chr5.fa.amb\n",
      "-rw-r--r-- 1 root root   44 Jul 24 08:59  chr5.fa.ann\n",
      "-rw-r--r-- 1 root root 174M Jul 24 08:59  chr5.fa.bwt\n",
      "-rw-r--r-- 1 root root   23 Jul 24 08:54  chr5.fa.fai\n",
      "-rw-r--r-- 1 root root  44M Jul 24 08:59  chr5.fa.pac\n",
      "-rw-r--r-- 1 root root  87M Jul 24 09:00  chr5.fa.sa\n",
      "-rw-r--r-- 1 root root 820K Jul 24 08:52  input.fq\n",
      "-rw-r--r-- 1 root root 225K Jul 24 08:54  input_fastqc.html\n",
      "-rw-r--r-- 1 root root 235K Jul 24 08:54  input_fastqc.zip\n",
      "-rw-r--r-- 1 root root 212K Jul 24 11:38  mapped.bam\n",
      "-rw-r--r-- 1 root root 214K Jul 24 11:42  mapped.dedup.bam\n",
      "-rw-r--r-- 1 root root 214K Jul 24 11:42  mapped.dedup.sort.bam\n",
      "-rw-r--r-- 1 root root  72K Jul 24 11:42  mapped.dedup.sort.bam.bai\n",
      "-rw-r--r-- 1 root root 963K Jul 24 11:06  mapped.sam\n"
     ]
    }
   ],
   "source": [
    "ls -lh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Indexing the reference fasta file\n",
    "\n",
    "To prepare the reference fasta file for variant calling, we need to index the file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [],
   "source": [
    "samtools faidx chr5.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 483M\n",
      "-rw-r--r-- 1 root root  15K Jul 24 09:02 '01 - Preparations for Finding a Disease Mutation.ipynb'\n",
      "-rw-r--r-- 1 root root  17K Jul 24 11:07 '02 - Aligning the FASTQ File.ipynb'\n",
      "-rw-r--r-- 1 root root  62K Jul 24 11:42 '03 - Variant Calling.ipynb'\n",
      "-rw-r--r-- 1 root root  11K Jul 24 08:52 '04 - Annotation of Variants.ipynb'\n",
      "-rw-r--r-- 1 root root 177M Jul 24 08:53  chr5.fa\n",
      "-rw-r--r-- 1 root root  588 Jul 24 08:59  chr5.fa.amb\n",
      "-rw-r--r-- 1 root root   44 Jul 24 08:59  chr5.fa.ann\n",
      "-rw-r--r-- 1 root root 174M Jul 24 08:59  chr5.fa.bwt\n",
      "-rw-r--r-- 1 root root   23 Jul 24 11:42  chr5.fa.fai\n",
      "-rw-r--r-- 1 root root  44M Jul 24 08:59  chr5.fa.pac\n",
      "-rw-r--r-- 1 root root  87M Jul 24 09:00  chr5.fa.sa\n",
      "-rw-r--r-- 1 root root 820K Jul 24 08:52  input.fq\n",
      "-rw-r--r-- 1 root root 225K Jul 24 08:54  input_fastqc.html\n",
      "-rw-r--r-- 1 root root 235K Jul 24 08:54  input_fastqc.zip\n",
      "-rw-r--r-- 1 root root 212K Jul 24 11:38  mapped.bam\n",
      "-rw-r--r-- 1 root root 214K Jul 24 11:42  mapped.dedup.bam\n",
      "-rw-r--r-- 1 root root 214K Jul 24 11:42  mapped.dedup.sort.bam\n",
      "-rw-r--r-- 1 root root  72K Jul 24 11:42  mapped.dedup.sort.bam.bai\n",
      "-rw-r--r-- 1 root root 963K Jul 24 11:06  mapped.sam\n"
     ]
    }
   ],
   "source": [
    "ls -lh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice the .fai file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr5\t181538259\t6\t50\t51\n"
     ]
    }
   ],
   "source": [
    "head chr5.fa.fai"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Variant calling using Platypus\n",
    "\n",
    "There are different methods for variant calling. Here, we will use Platypus, a haplotype-based variant caller that increases the sensitivity and specificity of the varianta calls (http://www.well.ox.ac.uk/platypus)\n",
    "\n",
    "<img src=\"https://bchdb.nus.edu.sg/media/notebook/ng.3036-F1.jpg\" width=800></img>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: freebayes [OPTION] ... [BAM FILE] ... \n",
      "\n",
      "Bayesian haplotype-based polymorphism discovery.\n",
      "\n",
      "citation: Erik Garrison, Gabor Marth\n",
      "          \"Haplotype-based variant detection from short-read sequencing\"\n",
      "          arXiv:1207.3907 (http://arxiv.org/abs/1207.3907)\n",
      "\n",
      "overview:\n",
      "\n",
      "    To call variants from aligned short-read sequencing data, supply BAM files and\n",
      "    a reference.  FreeBayes will provide VCF output on standard out describing SNPs,\n",
      "    indels, and complex variants in samples in the input alignments.\n",
      "\n",
      "    By default, FreeBayes will consider variants supported by at least 2\n",
      "    observations in a single sample (-C) and also by at least 20% of the reads from\n",
      "    a single sample (-F).  These settings are suitable to low to high depth\n",
      "    sequencing in haploid and diploid samples, but users working with polyploid or\n",
      "    pooled samples may wish to adjust them depending on the characteristics of\n",
      "    their sequencing data.\n",
      "\n",
      "    FreeBayes is capable of calling variant haplotypes shorter than a read length\n",
      "    where multiple polymorphisms segregate on the same read.  The maximum distance\n",
      "    between polymorphisms phased in this way is determined by the\n",
      "    --max-complex-gap, which defaults to 3bp.  In practice, this can comfortably be\n",
      "    set to half the read length.\n",
      "\n",
      "    Ploidy may be set to any level (-p), but by default all samples are assumed to\n",
      "    be diploid.  FreeBayes can model per-sample and per-region variation in\n",
      "    copy-number (-A) using a copy-number variation map.\n",
      "\n",
      "    FreeBayes can act as a frequency-based pooled caller and describe variants\n",
      "    and haplotypes in terms of observation frequency rather than called genotypes.\n",
      "    To do so, use --pooled-continuous and set input filters to a suitable level.\n",
      "    Allele observation counts will be described by AO and RO fields in the VCF output.\n",
      "\n",
      "\n",
      "examples:\n",
      "\n",
      "    # call variants assuming a diploid sample\n",
      "    freebayes -f ref.fa aln.bam >var.vcf\n",
      "\n",
      "    # call variants assuming a diploid sample, providing gVCF output\n",
      "    freebayes -f ref.fa --gvcf aln.bam >var.gvcf\n",
      "\n",
      "    # require at least 5 supporting observations to consider a variant\n",
      "    freebayes -f ref.fa -C 5 aln.bam >var.vcf\n",
      "\n",
      "    # discard alignments overlapping positions where total read depth is greater than 200\n",
      "    freebayes -f ref.fa -g 200 aln.bam >var.vcf\n",
      "\n",
      "    # use a different ploidy\n",
      "    freebayes -f ref.fa -p 4 aln.bam >var.vcf\n",
      "\n",
      "    # assume a pooled sample with a known number of genome copies\n",
      "    freebayes -f ref.fa -p 20 --pooled-discrete aln.bam >var.vcf\n",
      "\n",
      "    # generate frequency-based calls for all variants passing input thresholds\n",
      "    freebayes -f ref.fa -F 0.01 -C 1 --pooled-continuous aln.bam >var.vcf\n",
      "\n",
      "    # use an input VCF (bgzipped + tabix indexed) to force calls at particular alleles\n",
      "    freebayes -f ref.fa -@ in.vcf.gz aln.bam >var.vcf\n",
      "\n",
      "    # generate long haplotype calls over known variants\n",
      "    freebayes -f ref.fa --haplotype-basis-alleles in.vcf.gz \\ \n",
      "                        --haplotype-length 50 aln.bam\n",
      "\n",
      "    # naive variant calling: simply annotate observation counts of SNPs and indels\n",
      "    freebayes -f ref.fa --haplotype-length 0 --min-alternate-count 1 \\ \n",
      "        --min-alternate-fraction 0 --pooled-continuous --report-monomorphic >var.vcf\n",
      "\n",
      "\n",
      "parameters:\n",
      "\n",
      "   -h --help       Prints this help dialog.\n",
      "   --version       Prints the release number and the git commit id.\n",
      "\n",
      "input:\n",
      "\n",
      "   -b --bam FILE   Add FILE to the set of BAM files to be analyzed.\n",
      "   -L --bam-list FILE\n",
      "                   A file containing a list of BAM files to be analyzed.\n",
      "   -c --stdin      Read BAM input on stdin.\n",
      "   -f --fasta-reference FILE\n",
      "                   Use FILE as the reference sequence for analysis.\n",
      "                   An index file (FILE.fai) will be created if none exists.\n",
      "                   If neither --targets nor --region are specified, FreeBayes\n",
      "                   will analyze every position in this reference.\n",
      "   -t --targets FILE\n",
      "                   Limit analysis to targets listed in the BED-format FILE.\n",
      "   -r --region <chrom>:<start_position>-<end_position>\n",
      "                   Limit analysis to the specified region, 0-base coordinates,\n",
      "                   end_position not included (same as BED format).\n",
      "                   Either '-' or '..' maybe used as a separator.\n",
      "   -s --samples FILE\n",
      "                   Limit analysis to samples listed (one per line) in the FILE.\n",
      "                   By default FreeBayes will analyze all samples in its input\n",
      "                   BAM files.\n",
      "   --populations FILE\n",
      "                   Each line of FILE should list a sample and a population which\n",
      "                   it is part of.  The population-based bayesian inference model\n",
      "                   will then be partitioned on the basis of the populations.\n",
      "   -A --cnv-map FILE\n",
      "                   Read a copy number map from the BED file FILE, which has\n",
      "                   either a sample-level ploidy:\n",
      "                      sample_name copy_number\n",
      "                   or a region-specific format:\n",
      "                      seq_name start end sample_name copy_number\n",
      "                   ... for each region in each sample which does not have the\n",
      "                   default copy number as set by --ploidy. These fields can be delimited\n",
      "                   by space or tab.\n",
      "\n",
      "output:\n",
      "\n",
      "   -v --vcf FILE   Output VCF-format results to FILE. (default: stdout)\n",
      "   --gvcf\n",
      "                   Write gVCF output, which indicates coverage in uncalled regions.\n",
      "   --gvcf-chunk NUM\n",
      "                   When writing gVCF output emit a record for every NUM bases.\n",
      "   -& --gvcf-dont-use-chunk BOOL \n",
      "                   When writing the gVCF output emit a record for all bases if\n",
      "                   set to \"true\" , will also route an int to --gvcf-chunk\n",
      "                   similar to --output-mode EMIT_ALL_SITES from GATK\n",
      "   -@ --variant-input VCF\n",
      "                   Use variants reported in VCF file as input to the algorithm.\n",
      "                   Variants in this file will included in the output even if\n",
      "                   there is not enough support in the data to pass input filters.\n",
      "   -l --only-use-input-alleles\n",
      "                   Only provide variant calls and genotype likelihoods for sites\n",
      "                   and alleles which are provided in the VCF input, and provide\n",
      "                   output in the VCF for all input alleles, not just those which\n",
      "                   have support in the data.\n",
      "   --haplotype-basis-alleles VCF\n",
      "                   When specified, only variant alleles provided in this input\n",
      "                   VCF will be used for the construction of complex or haplotype\n",
      "                   alleles.\n",
      "   --report-all-haplotype-alleles\n",
      "                   At sites where genotypes are made over haplotype alleles,\n",
      "                   provide information about all alleles in output, not only\n",
      "                   those which are called.\n",
      "   --report-monomorphic\n",
      "                   Report even loci which appear to be monomorphic, and report all\n",
      "                   considered alleles, even those which are not in called genotypes.\n",
      "                   Loci which do not have any potential alternates have '.' for ALT.\n",
      "   -P --pvar N     Report sites if the probability that there is a polymorphism\n",
      "                   at the site is greater than N.  default: 0.0.  Note that post-\n",
      "                   filtering is generally recommended over the use of this parameter.\n",
      "   --strict-vcf\n",
      "                   Generate strict VCF format (FORMAT/GQ will be an int)\n",
      "\n",
      "population model:\n",
      "\n",
      "   -T --theta N    The expected mutation rate or pairwise nucleotide diversity\n",
      "                   among the population under analysis.  This serves as the\n",
      "                   single parameter to the Ewens Sampling Formula prior model\n",
      "                   default: 0.001\n",
      "   -p --ploidy N   Sets the default ploidy for the analysis to N.  default: 2\n",
      "   -J --pooled-discrete\n",
      "                   Assume that samples result from pooled sequencing.\n",
      "                   Model pooled samples using discrete genotypes across pools.\n",
      "                   When using this flag, set --ploidy to the number of\n",
      "                   alleles in each sample or use the --cnv-map to define\n",
      "                   per-sample ploidy.\n",
      "   -K --pooled-continuous\n",
      "                   Output all alleles which pass input filters, regardles of\n",
      "                   genotyping outcome or model.\n",
      "\n",
      "reference allele:\n",
      "\n",
      "   -Z --use-reference-allele\n",
      "                   This flag includes the reference allele in the analysis as\n",
      "                   if it is another sample from the same population.\n",
      "   --reference-quality MQ,BQ\n",
      "                   Assign mapping quality of MQ to the reference allele at each\n",
      "                   site and base quality of BQ.  default: 100,60\n",
      "\n",
      "allele scope:\n",
      "\n",
      "   -n --use-best-n-alleles N\n",
      "                   Evaluate only the best N SNP alleles, ranked by sum of\n",
      "                   supporting quality scores.  (Set to 0 to use all; default: all)\n",
      "   -E --max-complex-gap N\n",
      "      --haplotype-length N\n",
      "                   Allow haplotype calls with contiguous embedded matches of up\n",
      "                   to this length. Set N=-1 to disable clumping. (default: 3)\n",
      "   --min-repeat-size N\n",
      "                   When assembling observations across repeats, require the total repeat\n",
      "                   length at least this many bp.  (default: 5)\n",
      "   --min-repeat-entropy N\n",
      "                   To detect interrupted repeats, build across sequence until it has\n",
      "                   entropy > N bits per bp. Set to 0 to turn off. (default: 1)\n",
      "   --no-partial-observations\n",
      "                   Exclude observations which do not fully span the dynamically-determined\n",
      "                   detection window.  (default, use all observations, dividing partial\n",
      "                   support across matching haplotypes when generating haplotypes.)\n",
      "\n",
      "  These flags are meant for testing.\n",
      "  They are not meant for filtering the output.\n",
      "  They actually filter the input to the algorithm by throwing away alignments.\n",
      "  This hurts performance by hiding information from the Bayesian model.\n",
      "  Do not use them unless you can validate that they improve results!\n",
      "\n",
      "   -I --throw-away-snp-obs     Remove SNP observations from input.\n",
      "   -i --throw-away-indels-obs  Remove indel observations from input.\n",
      "   -X --throw-away-mnp-obs     Remove MNP observations from input.\n",
      "   -u --throw-away-complex-obs Remove complex allele observations from input.\n",
      "\n",
      "  If you need to break apart haplotype calls to obtain one class of alleles,\n",
      "  run the call with default parameters, then normalize and subset the VCF:\n",
      "    freebayes ... | vcfallelicprimitives -kg >calls.vcf\n",
      "  For example, this would retain only biallelic SNPs.\n",
      "    <calls.vcf vcfsnps | vcfbiallelic >biallelic_snp_calls.vcf\n",
      "\n",
      "indel realignment:\n",
      "\n",
      "   -O --dont-left-align-indels\n",
      "                   Turn off left-alignment of indels, which is enabled by default.\n",
      "\n",
      "input filters:\n",
      "\n",
      "   -4 --use-duplicate-reads\n",
      "                   Include duplicate-marked alignments in the analysis.\n",
      "                   default: exclude duplicates marked as such in alignments\n",
      "   -m --min-mapping-quality Q\n",
      "                   Exclude alignments from analysis if they have a mapping\n",
      "                   quality less than Q.  default: 1\n",
      "   -q --min-base-quality Q\n",
      "                   Exclude alleles from analysis if their supporting base\n",
      "                   quality is less than Q.  default: 0\n",
      "   -R --min-supporting-allele-qsum Q\n",
      "                   Consider any allele in which the sum of qualities of supporting\n",
      "                   observations is at least Q.  default: 0\n",
      "   -Y --min-supporting-mapping-qsum Q\n",
      "                   Consider any allele in which and the sum of mapping qualities of\n",
      "                   supporting reads is at least Q.  default: 0\n",
      "   -Q --mismatch-base-quality-threshold Q\n",
      "                   Count mismatches toward --read-mismatch-limit if the base\n",
      "                   quality of the mismatch is >= Q.  default: 10\n",
      "   -U --read-mismatch-limit N\n",
      "                   Exclude reads with more than N mismatches where each mismatch\n",
      "                   has base quality >= mismatch-base-quality-threshold.\n",
      "                   default: ~unbounded\n",
      "   -z --read-max-mismatch-fraction N\n",
      "                   Exclude reads with more than N [0,1] fraction of mismatches where\n",
      "                   each mismatch has base quality >= mismatch-base-quality-threshold\n",
      "                   default: 1.0\n",
      "   -$ --read-snp-limit N\n",
      "                   Exclude reads with more than N base mismatches, ignoring gaps\n",
      "                   with quality >= mismatch-base-quality-threshold.\n",
      "                   default: ~unbounded\n",
      "   -e --read-indel-limit N\n",
      "                   Exclude reads with more than N separate gaps.\n",
      "                   default: ~unbounded\n",
      "   -0 --standard-filters  Use stringent input base and mapping quality filters\n",
      "                   Equivalent to -m 30 -q 20 -R 0 -S 0\n",
      "   -F --min-alternate-fraction N\n",
      "                   Require at least this fraction of observations supporting\n",
      "                   an alternate allele within a single individual in the\n",
      "                   in order to evaluate the position.  default: 0.05\n",
      "   -C --min-alternate-count N\n",
      "                   Require at least this count of observations supporting\n",
      "                   an alternate allele within a single individual in order\n",
      "                   to evaluate the position.  default: 2\n",
      "   -3 --min-alternate-qsum N\n",
      "                   Require at least this sum of quality of observations supporting\n",
      "                   an alternate allele within a single individual in order\n",
      "                   to evaluate the position.  default: 0\n",
      "   -G --min-alternate-total N\n",
      "                   Require at least this count of observations supporting\n",
      "                   an alternate allele within the total population in order\n",
      "                   to use the allele in analysis.  default: 1\n",
      "   --min-coverage N\n",
      "                   Require at least this coverage to process a site. default: 0\n",
      "   --limit-coverage N\n",
      "                   Downsample per-sample coverage to this level if greater than this coverage.\n",
      "                   default: no limit\n",
      "   -g --skip-coverage N\n",
      "                   Skip processing of alignments overlapping positions with coverage >N.\n",
      "                   This filters sites above this coverage, but will also reduce data nearby.\n",
      "                   default: no limit\n",
      "\n",
      "population priors:\n",
      "\n",
      "   -k --no-population-priors\n",
      "                   Equivalent to --pooled-discrete --hwe-priors-off and removal of\n",
      "                   Ewens Sampling Formula component of priors.\n",
      "\n",
      "mappability priors:\n",
      "\n",
      "   -w --hwe-priors-off\n",
      "                   Disable estimation of the probability of the combination\n",
      "                   arising under HWE given the allele frequency as estimated\n",
      "                   by observation frequency.\n",
      "   -V --binomial-obs-priors-off\n",
      "                   Disable incorporation of prior expectations about observations.\n",
      "                   Uses read placement probability, strand balance probability,\n",
      "                   and read position (5'-3') probability.\n",
      "   -a --allele-balance-priors-off\n",
      "                   Disable use of aggregate probability of observation balance between alleles\n",
      "                   as a component of the priors.\n",
      "\n",
      "genotype likelihoods:\n",
      "\n",
      "   --observation-bias FILE\n",
      "                   Read length-dependent allele observation biases from FILE.\n",
      "                   The format is [length] [alignment efficiency relative to reference]\n",
      "                   where the efficiency is 1 if there is no relative observation bias.\n",
      "   --base-quality-cap Q\n",
      "                   Limit estimated observation quality by capping base quality at Q.\n",
      "   --prob-contamination F\n",
      "                   An estimate of contamination to use for all samples.  default: 10e-9\n",
      "   --legacy-gls    Use legacy (polybayes equivalent) genotype likelihood calculations\n",
      "   --contamination-estimates FILE\n",
      "                   A file containing per-sample estimates of contamination, such as\n",
      "                   those generated by VerifyBamID.  The format should be:\n",
      "                       sample p(read=R|genotype=AR) p(read=A|genotype=AA)\n",
      "                   Sample '*' can be used to set default contamination estimates.\n",
      "\n",
      "algorithmic features:\n",
      "\n",
      "   --report-genotype-likelihood-max\n",
      "                   Report genotypes using the maximum-likelihood estimate provided\n",
      "                   from genotype likelihoods.\n",
      "   -B --genotyping-max-iterations N\n",
      "                   Iterate no more than N times during genotyping step. default: 1000.\n",
      "   --genotyping-max-banddepth N\n",
      "                   Integrate no deeper than the Nth best genotype by likelihood when\n",
      "                   genotyping. default: 6.\n",
      "   -W --posterior-integration-limits N,M\n",
      "                   Integrate all genotype combinations in our posterior space\n",
      "                   which include no more than N samples with their Mth best\n",
      "                   data likelihood. default: 1,3.\n",
      "   -N --exclude-unobserved-genotypes\n",
      "                   Skip sample genotypings for which the sample has no supporting reads.\n",
      "   -S --genotype-variant-threshold N\n",
      "                   Limit posterior integration to samples where the second-best\n",
      "                   genotype likelihood is no more than log(N) from the highest\n",
      "                   genotype likelihood for the sample.  default: ~unbounded\n",
      "   -j --use-mapping-quality\n",
      "                   Use mapping quality of alleles when calculating data likelihoods.\n",
      "   -H --harmonic-indel-quality\n",
      "                   Use a weighted sum of base qualities around an indel, scaled by the\n",
      "                   distance from the indel.  By default use a minimum BQ in flanking sequence.\n",
      "   -D --read-dependence-factor N\n",
      "                   Incorporate non-independence of reads by scaling successive\n",
      "                   observations by this factor during data likelihood\n",
      "                   calculations.  default: 0.9\n",
      "   -= --genotype-qualities\n",
      "                   Calculate the marginal probability of genotypes and report as GQ in\n",
      "                   each sample field in the VCF output.\n",
      "\n",
      "debugging:\n",
      "\n",
      "   -d --debug      Print debugging output.\n",
      "   -dd             Print more verbose debugging output (requires \"make DEBUG\")\n",
      "\n",
      "\n",
      "author:   Erik Garrison <erik.garrison@gmail.com>\n",
      "version:  v1.3.2-dirty\n"
     ]
    }
   ],
   "source": [
    "freebayes -h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The variant caller requires the reference file and the BAM file with its index."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [],
   "source": [
    "freebayes -f chr5.fa mapped.dedup.sort.bam > result.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's look at the first and last 50 lines of the VCF file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false,
    "jupyter": {
     "outputs_hidden": false
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##fileformat=VCFv4.2\n",
      "##fileDate=20200724\n",
      "##source=freeBayes v1.3.2-dirty\n",
      "##reference=chr5.fa\n",
      "##contig=<ID=chr5,length=181538259>\n",
      "##phasing=none\n",
      "##commandline=\"freebayes -f chr5.fa mapped.dedup.sort.bam\"\n",
      "##INFO=<ID=NS,Number=1,Type=Integer,Description=\"Number of samples with data\">\n",
      "##INFO=<ID=DP,Number=1,Type=Integer,Description=\"Total read depth at the locus\">\n",
      "##INFO=<ID=DPB,Number=1,Type=Float,Description=\"Total read depth per bp at the locus; bases in reads overlapping / bases in haplotype\">\n",
      "##INFO=<ID=AC,Number=A,Type=Integer,Description=\"Total number of alternate alleles in called genotypes\">\n",
      "##INFO=<ID=AN,Number=1,Type=Integer,Description=\"Total number of alleles in called genotypes\">\n",
      "##INFO=<ID=AF,Number=A,Type=Float,Description=\"Estimated allele frequency in the range (0,1]\">\n",
      "##INFO=<ID=RO,Number=1,Type=Integer,Description=\"Count of full observations of the reference haplotype.\">\n",
      "##INFO=<ID=AO,Number=A,Type=Integer,Description=\"Count of full observations of this alternate haplotype.\">\n",
      "##INFO=<ID=PRO,Number=1,Type=Float,Description=\"Reference allele observation count, with partial observations recorded fractionally\">\n",
      "##INFO=<ID=PAO,Number=A,Type=Float,Description=\"Alternate allele observations, with partial observations recorded fractionally\">\n",
      "##INFO=<ID=QR,Number=1,Type=Integer,Description=\"Reference allele quality sum in phred\">\n",
      "##INFO=<ID=QA,Number=A,Type=Integer,Description=\"Alternate allele quality sum in phred\">\n",
      "##INFO=<ID=PQR,Number=1,Type=Float,Description=\"Reference allele quality sum in phred for partial observations\">\n",
      "##INFO=<ID=PQA,Number=A,Type=Float,Description=\"Alternate allele quality sum in phred for partial observations\">\n",
      "##INFO=<ID=SRF,Number=1,Type=Integer,Description=\"Number of reference observations on the forward strand\">\n",
      "##INFO=<ID=SRR,Number=1,Type=Integer,Description=\"Number of reference observations on the reverse strand\">\n",
      "##INFO=<ID=SAF,Number=A,Type=Integer,Description=\"Number of alternate observations on the forward strand\">\n",
      "##INFO=<ID=SAR,Number=A,Type=Integer,Description=\"Number of alternate observations on the reverse strand\">\n",
      "##INFO=<ID=SRP,Number=1,Type=Float,Description=\"Strand balance probability for the reference allele: Phred-scaled upper-bounds estimate of the probability of observing the deviation between SRF and SRR given E(SRF/SRR) ~ 0.5, derived using Hoeffding's inequality\">\n",
      "##INFO=<ID=SAP,Number=A,Type=Float,Description=\"Strand balance probability for the alternate allele: Phred-scaled upper-bounds estimate of the probability of observing the deviation between SAF and SAR given E(SAF/SAR) ~ 0.5, derived using Hoeffding's inequality\">\n",
      "##INFO=<ID=AB,Number=A,Type=Float,Description=\"Allele balance at heterozygous sites: a number between 0 and 1 representing the ratio of reads showing the reference allele to all reads, considering only reads from individuals called as heterozygous\">\n",
      "##INFO=<ID=ABP,Number=A,Type=Float,Description=\"Allele balance probability at heterozygous sites: Phred-scaled upper-bounds estimate of the probability of observing the deviation between ABR and ABA given E(ABR/ABA) ~ 0.5, derived using Hoeffding's inequality\">\n",
      "##INFO=<ID=RUN,Number=A,Type=Integer,Description=\"Run length: the number of consecutive repeats of the alternate allele in the reference genome\">\n",
      "##INFO=<ID=RPP,Number=A,Type=Float,Description=\"Read Placement Probability: Phred-scaled upper-bounds estimate of the probability of observing the deviation between RPL and RPR given E(RPL/RPR) ~ 0.5, derived using Hoeffding's inequality\">\n",
      "##INFO=<ID=RPPR,Number=1,Type=Float,Description=\"Read Placement Probability for reference observations: Phred-scaled upper-bounds estimate of the probability of observing the deviation between RPL and RPR given E(RPL/RPR) ~ 0.5, derived using Hoeffding's inequality\">\n",
      "##INFO=<ID=RPL,Number=A,Type=Float,Description=\"Reads Placed Left: number of reads supporting the alternate balanced to the left (5') of the alternate allele\">\n",
      "##INFO=<ID=RPR,Number=A,Type=Float,Description=\"Reads Placed Right: number of reads supporting the alternate balanced to the right (3') of the alternate allele\">\n",
      "##INFO=<ID=EPP,Number=A,Type=Float,Description=\"End Placement Probability: Phred-scaled upper-bounds estimate of the probability of observing the deviation between EL and ER given E(EL/ER) ~ 0.5, derived using Hoeffding's inequality\">\n",
      "##INFO=<ID=EPPR,Number=1,Type=Float,Description=\"End Placement Probability for reference observations: Phred-scaled upper-bounds estimate of the probability of observing the deviation between EL and ER given E(EL/ER) ~ 0.5, derived using Hoeffding's inequality\">\n",
      "##INFO=<ID=DPRA,Number=A,Type=Float,Description=\"Alternate allele depth ratio.  Ratio between depth in samples with each called alternate allele and those without.\">\n",
      "##INFO=<ID=ODDS,Number=1,Type=Float,Description=\"The log odds ratio of the best genotype combination to the second-best.\">\n",
      "##INFO=<ID=GTI,Number=1,Type=Integer,Description=\"Number of genotyping iterations required to reach convergence or bailout.\">\n",
      "##INFO=<ID=TYPE,Number=A,Type=String,Description=\"The type of allele, either snp, mnp, ins, del, or complex.\">\n",
      "##INFO=<ID=CIGAR,Number=A,Type=String,Description=\"The extended CIGAR representation of each alternate allele, with the exception that '=' is replaced by 'M' to ease VCF parsing.  Note that INDEL alleles do not have the first matched base (which is provided by default, per the spec) referred to by the CIGAR.\">\n",
      "##INFO=<ID=NUMALT,Number=1,Type=Integer,Description=\"Number of unique non-reference alleles in called genotypes at this position.\">\n",
      "##INFO=<ID=MEANALT,Number=A,Type=Float,Description=\"Mean number of unique non-reference allele observations per sample with the corresponding alternate alleles.\">\n",
      "##INFO=<ID=LEN,Number=A,Type=Integer,Description=\"allele length\">\n",
      "##INFO=<ID=MQM,Number=A,Type=Float,Description=\"Mean mapping quality of observed alternate alleles\">\n",
      "##INFO=<ID=MQMR,Number=1,Type=Float,Description=\"Mean mapping quality of observed reference alleles\">\n",
      "##INFO=<ID=PAIRED,Number=A,Type=Float,Description=\"Proportion of observed alternate alleles which are supported by properly paired read fragments\">\n",
      "##INFO=<ID=PAIREDR,Number=1,Type=Float,Description=\"Proportion of observed reference alleles which are supported by properly paired read fragments\">\n",
      "##INFO=<ID=MIN_DP,Number=1,Type=Integer,Description=\"Minimum depth in gVCF output block.\">\n",
      "##INFO=<ID=END,Number=1,Type=Integer,Description=\"Last position (inclusive) in gVCF output record.\">\n",
      "##FORMAT=<ID=GT,Number=1,Type=String,Description=\"Genotype\">\n",
      "##FORMAT=<ID=GQ,Number=1,Type=Float,Description=\"Genotype Quality, the Phred-scaled marginal (or unconditional) probability of the called genotype\">\n",
      "##FORMAT=<ID=GL,Number=G,Type=Float,Description=\"Genotype Likelihood, log10-scaled likelihoods of the data given the called genotype for each possible genotype generated from the reference and alternate alleles given the sample ploidy\">\n",
      "##FORMAT=<ID=DP,Number=1,Type=Integer,Description=\"Read Depth\">\n",
      "##FORMAT=<ID=AD,Number=R,Type=Integer,Description=\"Number of observation for each allele\">\n",
      "##FORMAT=<ID=RO,Number=1,Type=Integer,Description=\"Reference allele observation count\">\n",
      "##FORMAT=<ID=QR,Number=1,Type=Integer,Description=\"Sum of quality of the reference observations\">\n",
      "##FORMAT=<ID=AO,Number=A,Type=Integer,Description=\"Alternate allele observation count\">\n",
      "##FORMAT=<ID=QA,Number=A,Type=Integer,Description=\"Sum of quality of the alternate observations\">\n",
      "##FORMAT=<ID=MIN_DP,Number=1,Type=Integer,Description=\"Minimum depth in gVCF output block.\">\n",
      "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tunknown\n",
      "chr5\t149004778\t.\tT\tC\t0\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=34;DPB=34;DPRA=0;EPP=7.35324;EPPR=4.09604;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=48.6934;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=4;QR=1054;RO=32;RPL=2;RPP=7.35324;RPPR=4.09604;RPR=0;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=32;SRP=72.4974;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:34:32,2:32:1054:2:4:0,-9.85502,-94.7643\n",
      "chr5\t149006962\t.\tT\tG\t360.547\t.\tAB=0.404762;ABP=6.31921;AC=1;AF=0.5;AN=2;AO=17;CIGAR=1X;DP=42;DPB=42;DPRA=0;EPP=6.20364;EPPR=3.79203;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=83.019;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=653;QR=946;RO=25;RPL=6;RPP=6.20364;RPPR=3.79203;RPR=11;RUN=1;SAF=17;SAP=39.9253;SAR=0;SRF=25;SRP=57.2971;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/1:42:25,17:25:946:17:653:-46.4593,0,-72.8018\n",
      "chr5\t149008881\t.\tT\tC\t3.46791e-16\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=33;DPB=33;DPRA=0;EPP=7.35324;EPPR=4.16842;GTI=0;LEN=1;MEANALT=2;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=45.0154;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=9;QR=760;RO=30;RPL=2;RPP=7.35324;RPPR=4.16842;RPR=0;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=30;SRP=68.1545;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:33:30,2:30:760:2:9:0,-8.80629,-67.7768\n",
      "chr5\t149010200\t.\tG\tA\t382.364\t.\tAB=0.515152;ABP=3.0761;AC=1;AF=0.5;AN=2;AO=17;CIGAR=1X;DP=33;DPB=33;DPRA=0;EPP=18.4661;EPPR=3.0103;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=79.616;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=671;QR=592;RO=16;RPL=3;RPP=18.4661;RPPR=3.0103;RPR=14;RUN=1;SAF=17;SAP=39.9253;SAR=0;SRF=16;SRP=37.7539;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/1:33:16,17:16:592:17:671:-50.7909,0,-43.6741\n",
      "chr5\t149010305\t.\tTCCCA\tGCCCC,GCCCA\t0.00400031\t.\tAB=0.4375,0.28125;ABP=4.09604,16.3106;AC=0,1;AF=0,0.5;AN=2;AO=14,9;CIGAR=1X3M1X,1X4M;DP=32;DPB=32.4;DPRA=0,0;EPP=25.3454,9.04217;EPPR=10.7656;GTI=0;LEN=5,1;MEANALT=4,4;MQM=60,60;MQMR=60;NS=1;NUMALT=2;ODDS=6.98948;PAIRED=0,0;PAIREDR=0;PAO=1,1;PQA=2,2;PQR=0;PRO=0;QA=112,106;QR=220;RO=7;RPL=13,7;RPP=25.3454,9.04217;RPPR=10.7656;RPR=1,2;RUN=1,1;SAF=14,9;SAP=33.4109,22.5536;SAR=0,0;SRF=7;SRP=18.2106;SRR=0;TYPE=complex,snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/2:32:7,14,9:7:220:14,9:112,106:-4.73777,-0.992538,-14.5394,0,-11.9918,-15.0428\n",
      "chr5\t149010310\t.\tT\tA\t0\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=34;DPB=34;DPRA=0;EPP=7.35324;EPPR=16.3106;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=59.75;NS=1;NUMALT=1;ODDS=44.1614;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=7;QR=355;RO=32;RPL=2;RPP=7.35324;RPPR=16.3106;RPR=0;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=32;SRP=72.4974;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:34:32,2:32:355:2:7:0,-9.57002,-31.3759\n",
      "chr5\t149010312\t.\tG\tT\t4.31991e-14\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=3;CIGAR=1X;DP=33;DPB=33;DPRA=0;EPP=3.73412;EPPR=21.5402;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=59.7333;NS=1;NUMALT=1;ODDS=32.9446;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=18;QR=318;RO=30;RPL=2;RPP=3.73412;RPPR=21.5402;RPR=1;RUN=1;SAF=3;SAP=9.52472;SAR=0;SRF=30;SRP=68.1545;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:33:30,3:30:318:3:18:0,-8.254,-27.0279\n",
      "chr5\t149010313\t.\tA\tT\t4.33347e-14\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=32;DPB=32;DPRA=0;EPP=3.0103;EPPR=19.8579;GTI=0;LEN=1;MEANALT=2;MQM=60;MQMR=59.7241;NS=1;NUMALT=1;ODDS=38.6992;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=7;QR=333;RO=29;RPL=1;RPP=3.0103;RPPR=19.8579;RPR=1;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=29;SRP=65.983;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:32:29,2:29:333:2:7:0,-8.69193,-29.4057\n",
      "chr5\t149010318\t.\tC\tA\t1.79152e-14\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=29;DPB=29;DPRA=0;EPP=7.35324;EPPR=8.35546;GTI=0;LEN=1;MEANALT=2;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=40.8165;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=4;QR=372;RO=26;RPL=2;RPP=7.35324;RPPR=8.35546;RPR=0;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=26;SRP=59.4686;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:29:26,2:26:372:2:4:0,-8.06884,-33.2154\n",
      "chr5\t149010320\t.\tC\tA\t1.76519e-14\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=3;CIGAR=1X;DP=28;DPB=28;DPRA=0;EPP=9.52472;EPPR=7.26639;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=59.68;NS=1;NUMALT=1;ODDS=40.2759;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=6;QR=405;RO=25;RPL=3;RPP=9.52472;RPPR=7.26639;RPR=0;RUN=1;SAF=3;SAP=9.52472;SAR=0;SRF=25;SRP=57.2971;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:28:25,3:25:405:3:6:0,-7.86884,-36.0268\n",
      "chr5\t149010321\t.\tA\tC\t7.83587e-15\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=28;DPB=28;DPRA=0;EPP=7.35324;EPPR=8.35546;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=59.6923;NS=1;NUMALT=1;ODDS=40.7704;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=4;QR=384;RO=26;RPL=2;RPP=7.35324;RPPR=8.35546;RPR=0;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=26;SRP=59.4686;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:28:26,2:26:384:2:4:0,-8.04884,-34.3029\n",
      "chr5\t149010322\t.\tT\tA\t2.07935e-14\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=7;CIGAR=1X;DP=27;DPB=27;DPRA=0;EPP=10.7656;EPPR=4.03889;GTI=0;LEN=1;MEANALT=2;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=35.0152;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=14;QR=337;RO=19;RPL=6;RPP=10.7656;RPPR=4.03889;RPR=1;RUN=1;SAF=7;SAP=18.2106;SAR=0;SRF=19;SRP=44.2683;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:27:19,7:19:337:7:14:0,-6.56678,-29.1943\n",
      "chr5\t149010324\t.\tA\tC\t1.94902e-14\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=28;DPB=28;DPRA=0;EPP=7.35324;EPPR=6.01695;GTI=0;LEN=1;MEANALT=1;MQM=56;MQMR=60;NS=1;NUMALT=1;ODDS=40.7704;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=4;QR=465;RO=26;RPL=2;RPP=7.35324;RPPR=6.01695;RPR=0;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=26;SRP=59.4686;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:28:26,2:26:465:2:4:0,-8.04884,-41.6153\n",
      "chr5\t149010327\t.\tT\tC\t2.1999e-14\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=4;CIGAR=1X;DP=26;DPB=26;DPRA=0;EPP=11.6962;EPPR=3.40511;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=59.6364;NS=1;NUMALT=1;ODDS=37.6502;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=8;QR=402;RO=22;RPL=4;RPP=11.6962;RPPR=3.40511;RPR=0;RUN=1;SAF=4;SAP=11.6962;SAR=0;SRF=22;SRP=50.7827;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:26:22,4:22:402:4:8:0,-7.08678,-35.5951\n",
      "chr5\t149010336\t.\tA\tC\t1.94448e-14\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=3;CIGAR=1X;DP=24;DPB=24;DPRA=0;EPP=9.52472;EPPR=3.1137;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=35.2123;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=6;QR=437;RO=21;RPL=3;RPP=9.52472;RPPR=3.1137;RPR=0;RUN=1;SAF=3;SAP=9.52472;SAR=0;SRF=21;SRP=48.6112;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:24:21,3:21:437:3:6:0,-6.66472,-38.9489\n",
      "chr5\t149010345\t.\tG\tT\t2.63827e-13\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=4;CIGAR=1X;DP=20;DPB=20;DPRA=0;EPP=11.6962;EPPR=7.89611;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=30.4592;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=8;QR=466;RO=16;RPL=4;RPP=11.6962;RPPR=7.89611;RPR=0;RUN=1;SAF=4;SAP=11.6962;SAR=0;SRF=16;SRP=37.7539;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:20:16,4:16:466:4:8:0,-5.2806,-41.4614\n",
      "chr5\t149010364\t.\tT\tC\t1.28923e-11\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=17;DPB=17;DPRA=0;EPP=7.35324;EPPR=3.15506;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=26.5434;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=4;QR=565;RO=15;RPL=2;RPP=7.35324;RPPR=3.15506;RPR=0;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=15;SRP=35.5824;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:17:15,2:15:565:2:4:0,-4.73751,-50.8062\n",
      "chr5\t149012615\t.\tT\tC\t1.58234e-15\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=3;CIGAR=1X;DP=47;DPB=47;DPRA=0;EPP=3.73412;EPPR=6.16881;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=54.0107;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=47;QR=1547;RO=44;RPL=2;RPP=3.73412;RPPR=6.16881;RPR=1;RUN=1;SAF=3;SAP=9.52472;SAR=0;SRF=44;SRP=98.5551;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:47:44,3:44:1547:3:47:0,-9.76303,-135.102\n",
      "chr5\t149026469\t.\tC\tT\t105.32\t.\tAB=0;ABP=0;AC=2;AF=1;AN=2;AO=4;CIGAR=1X;DP=4;DPB=4;DPRA=0;EPP=5.18177;EPPR=0;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=0;NS=1;NUMALT=1;ODDS=10.1503;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=140;QR=0;RO=0;RPL=1;RPP=5.18177;RPPR=0;RPR=3;RUN=1;SAF=4;SAP=11.6962;SAR=0;SRF=0;SRP=0;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t1/1:4:0,4:0:0:4:140:-12.9402,-1.20412,0\n",
      "chr5\t149026671\t.\tT\tC\t1.83057e-08\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=25;DPB=25;DPRA=0;EPP=3.0103;EPPR=3.86001;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=19.2846;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=78;QR=799;RO=23;RPL=1;RPP=3.0103;RPPR=3.86001;RPR=1;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=23;SRP=52.9542;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:25:23,2:23:799:2:78:0,-0.122449,-64.8072\n",
      "chr5\t149026823\t.\tT\tC\t260.732\t.\tAB=0.48;ABP=3.09716;AC=1;AF=0.5;AN=2;AO=12;CIGAR=1X;DP=25;DPB=25;DPRA=0;EPP=9.52472;EPPR=7.18621;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=60.0358;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=469;QR=506;RO=13;RPL=3;RPP=9.52472;RPPR=7.18621;RPR=9;RUN=1;SAF=12;SAP=29.068;SAR=0;SRF=13;SRP=31.2394;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/1:25:13,12:13:506:12:469:-35.0352,0,-38.3583\n",
      "chr5\t149026872\t.\tG\tA\t358.298\t.\tAB=0.377778;ABP=8.84915;AC=1;AF=0.5;AN=2;AO=17;CIGAR=1X;DP=45;DPB=45;DPRA=0;EPP=3.13803;EPPR=10.7656;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=82.5012;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=657;QR=1065;RO=28;RPL=8;RPP=3.13803;RPPR=10.7656;RPR=9;RUN=1;SAF=17;SAP=39.9253;SAR=0;SRF=28;SRP=63.8115;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/1:45:28,17:28:1065:17:657:-45.9145,0,-82.6057\n",
      "chr5\t149027190\t.\tC\tA\t1.4601e-14\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=26;DPB=26;DPRA=0;EPP=7.35324;EPPR=3.37221;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=37.2739;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=8;QR=899;RO=24;RPL=2;RPP=7.35324;RPPR=3.37221;RPR=0;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=24;SRP=55.1256;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:26:24,2:24:899:2:8:0,-7.06678,-80.4578\n",
      "chr5\t149027678\t.\tA\tG\t5.57478e-15\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=39;DPB=39;DPRA=0;EPP=7.35324;EPPR=7.76406;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=47.4777;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=40;QR=1252;RO=37;RPL=2;RPP=7.35324;RPPR=7.76406;RPR=0;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=37;SRP=83.3548;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:39:37,2:37:1252:2:40:0,-7.94276,-109.153\n",
      "chr5\t149027738\t.\tT\tG\t3.30621e-11\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=19;DPB=19;DPRA=0;EPP=7.35324;EPPR=5.18177;GTI=0;LEN=1;MEANALT=2;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=25.6015;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=15;QR=273;RO=16;RPL=0;RPP=7.35324;RPPR=5.18177;RPR=2;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=16;SRP=37.7539;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:19:16,2:16:273:2:15:0,-4.07854,-23.3049\n",
      "chr5\t149027786\t.\tT\tG\t5.72201e-07\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=11;DPB=11;DPRA=0;EPP=7.35324;EPPR=7.35324;GTI=0;LEN=1;MEANALT=2;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=15.8423;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=14;QR=172;RO=8;RPL=2;RPP=7.35324;RPPR=7.35324;RPR=0;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=8;SRP=20.3821;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:11:8,2:8:172:2:14:0,-1.71697,-14.3526\n",
      "chr5\t149028145\t.\tA\tC\t273.868\t.\tAB=0;ABP=0;AC=2;AF=1;AN=2;AO=10;CIGAR=1X;DP=12;DPB=12;DPRA=0;EPP=6.48466;EPPR=7.35324;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=20.3349;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=339;QR=4;RO=2;RPL=7;RPP=6.48466;RPPR=7.35324;RPR=3;RUN=1;SAF=10;SAP=24.725;SAR=0;SRF=2;SRP=7.35324;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t1/1:12:2,10:2:4:10:339:-30.4531,-3.23236,0\n",
      "chr5\t149028162\t.\tT\tC\t7.37749e-07\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=8;DPB=8;DPRA=0;EPP=7.35324;EPPR=8.80089;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=15.6472;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=4;QR=114;RO=6;RPL=2;RPP=7.35324;RPPR=8.80089;RPR=0;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=6;SRP=16.0391;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:8:6,2:6:114:2:4:0,-2.02824,-10.0678\n",
      "chr5\t149028435\t.\tA\tC\t1.28923e-11\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=17;DPB=17;DPRA=0;EPP=7.35324;EPPR=4.31318;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=26.5434;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=4;QR=473;RO=15;RPL=2;RPP=7.35324;RPPR=4.31318;RPR=0;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=15;SRP=35.5824;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:17:15,2:15:473:2:4:0,-4.73751,-42.4876\n",
      "chr5\t149028538\t.\tA\tG\t6.51745\t.\tAB=0.214286;ABP=12.937;AC=1;AF=0.5;AN=2;AO=3;CIGAR=1X;DP=14;DPB=14;DPRA=0;EPP=3.73412;EPPR=4.78696;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=1.24842;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=109;QR=368;RO=11;RPL=1;RPP=3.73412;RPPR=4.78696;RPR=2;RUN=1;SAF=3;SAP=9.52472;SAR=0;SRF=11;SRP=26.8965;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/1:14:11,3:11:368:3:109:-5.95053,0,-29.2129\n",
      "chr5\t149040507\t.\tA\tG\t2.33808e-10\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=27;DPB=27;DPRA=0;EPP=3.0103;EPPR=7.26639;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=23.6451;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=70;QR=966;RO=25;RPL=1;RPP=3.0103;RPPR=7.26639;RPR=1;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=25;SRP=57.2971;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:27:25,2:25:966:2:70:0,-1.48233,-80.6003\n",
      "chr5\t149041410\t.\tA\tG\t6.32863e-15\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=38;DPB=38;DPRA=0;EPP=3.0103;EPPR=6.8707;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=43.4453;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=46;QR=1275;RO=36;RPL=1;RPP=3.0103;RPPR=6.8707;RPR=1;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=36;SRP=81.1833;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:38:36,2:36:1275:2:46:0,-7.07174,-110.645\n",
      "chr5\t149041580\t.\tT\tG\t0\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=8;CIGAR=1X;DP=52;DPB=52;DPRA=0;EPP=7.35324;EPPR=14.3727;GTI=0;LEN=1;MEANALT=2;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=41.1053;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=115;QR=1101;RO=43;RPL=6;RPP=7.35324;RPPR=14.3727;RPR=2;RUN=1;SAF=8;SAP=20.3821;SAR=0;SRF=43;SRP=96.3836;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:52:43,8:43:1101:8:115:0,-4.97285,-88.8342\n",
      "chr5\t149041583\t.\tA\tG\t1.11547e-15\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=4;CIGAR=1X;DP=51;DPB=51;DPRA=0;EPP=3.0103;EPPR=20.4303;GTI=0;LEN=1;MEANALT=2;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=53.7066;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=54;QR=1165;RO=45;RPL=2;RPP=3.0103;RPPR=20.4303;RPR=2;RUN=1;SAF=4;SAP=11.6962;SAR=0;SRF=45;SRP=100.727;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:51:45,4:45:1165:4:54:0,-9.84402,-100.112\n",
      "chr5\t149041619\t.\tT\tC\t0\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=30;DPB=30;DPRA=0;EPP=7.35324;EPPR=10.7656;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=43.4026;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=4;QR=883;RO=28;RPL=2;RPP=7.35324;RPPR=10.7656;RPR=0;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=28;SRP=63.8115;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:30:28,2:28:883:2:4:0,-8.6509,-79.3642\n",
      "chr5\t149042718\t.\tA\tG\t681.474\t.\tAB=0.491525;ABP=3.0471;AC=1;AF=0.5;AN=2;AO=29;CIGAR=1X;DP=59;DPB=59;DPRA=0;EPP=6.67934;EPPR=4.16842;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=156.915;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=1108;QR=1133;RO=30;RPL=18;RPP=6.67934;RPPR=4.16842;RPR=11;RUN=1;SAF=29;SAP=65.983;SAR=0;SRF=30;SRP=68.1545;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/1:59:30,29:30:1133:29:1108:-82.2479,0,-84.5084\n",
      "chr5\t149063022\t.\tT\tC\t132.755\t.\tAB=0.392857;ABP=5.80219;AC=1;AF=0.5;AN=2;AO=11;CIGAR=1X;DP=28;DPB=28;DPRA=0;EPP=12.6832;EPPR=6.20364;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=30.568;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=344;QR=558;RO=17;RPL=9;RPP=12.6832;RPPR=6.20364;RPR=2;RUN=1;SAF=11;SAP=26.8965;SAR=0;SRF=17;SRP=39.9253;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/1:28:17,11:17:558:11:344:-22.8257,0,-42.0909\n",
      "chr5\t149099303\t.\tT\tC\t57.8588\t.\tAB=0;ABP=0;AC=2;AF=1;AN=2;AO=2;CIGAR=1X;DP=2;DPB=2;DPRA=0;EPP=7.35324;EPPR=0;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=0;NS=1;NUMALT=1;ODDS=7.37776;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=80;QR=0;RO=0;RPL=0;RPP=7.35324;RPPR=0;RPR=2;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=0;SRP=0;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t1/1:2:0,2:0:0:2:80:-7.59179,-0.60206,0\n",
      "chr5\t149142161\t.\tT\tG\t8.54694e-10\t.\tAB=0;ABP=0;AC=0;AF=0;AN=2;AO=2;CIGAR=1X;DP=14;DPB=14;DPRA=0;EPP=7.35324;EPPR=21.1059;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=1;NUMALT=1;ODDS=22.3488;PAIRED=0;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=6;QR=285;RO=12;RPL=2;RPP=7.35324;RPPR=21.1059;RPR=0;RUN=1;SAF=2;SAP=7.35324;SAR=0;SRF=12;SRP=29.068;SRR=0;TYPE=snp\tGT:DP:AD:RO:QR:AO:QA:GL\t0/0:14:12,2:12:285:2:6:0,-3.64442,-25.3149\n"
     ]
    }
   ],
   "source": [
    "head -n 50 result.vcf\n",
    "tail -n 50 result.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
